src/blockdata/transaction.rs

```diff
@@ -262,6 +255,58 @@ impl Default for TxOut {
     }
 }
 
+/// Result of [`Transaction::encode_signing_data_to`].
+///
+/// This type forces the caller to handle SIGHASH_SINGLE bug corner case.
+pub enum EncodeSigningDataResult<E> {
+    /// Input data is an instance of `SIGHASH_SINGLE` bug
+    SighashSingleBug,
+    /// Operation performed normally.
+    Result(Result<(), E>),
```

Comment:
I don't like naming custom items with names like `Result`, `Ok` because of confusion. This case is pretty reasonable but consider naming it `WriteResult` or something else. If we can't figure out a better name, it's fine.